File: DSA/Asmi_Garg/day8_ques2.java
import java.util.LinkedList;
import java.util.Queue;

class StackUsingTwoQueues {
    private Queue<Integer> queue1;
    private Queue<Integer> queue2;

    public StackUsingTwoQueues() {
        queue1 = new LinkedList<>();
        queue2 = new LinkedList<>();
    }

    public void push(int x) {
        queue2.offer(x);

        while (!queue1.isEmpty()) {
            queue2.offer(queue1.poll());
        }

        Queue<Integer> temp = queue1;
        queue1 = queue2;
        queue2 = temp;
    }

    public int pop() {
        if (queue1.isEmpty()) {
            throw new RuntimeException("Stack is empty");
        }
        return queue1.poll();
    }

    public int top() {
        if (queue1.isEmpty()) {
            throw new RuntimeException("Stack is empty");
        }
        return queue1.peek();
    }

    public boolean isEmpty() {
        return queue1.isEmpty();
    }

    public static void main(String[] args) {
        StackUsingTwoQueues stack = new StackUsingTwoQueues();

        stack.push(1);
        stack.push(2);

        System.out.println("Top element: " + stack.top());
        System.out.println("Popped element: " + stack.pop());
        System.out.println("Top element after pop: " + stack.top());
    }
}

File: DSA/Anisha_Parveen/Day_8_Ques_2.cpp
#include <bits/stdc++.h>
using namespace std;

queue<int> q1, q2;

void push(int x)
{
    q1.push(x);
}

int pop()
{
    while (q1.size() > 1)
    {
        q2.push(q1.front());
        q1.pop();
    }

    int element = q1.front();
    q1.pop();

    swap(q1, q2);

    return element;
}
int top()
{
    while (q1.size() > 1)
    {
        q2.push(q1.front());
        q1.pop();
    }
    int element = q1.front();

    q2.push(element);
    swap(q1, q2);

    return element;
}

bool empty()
{
    return q1.empty();
}

int main()
{
    push(1);
    push(2);

    cout << "Top Operation: " << top() << endl;
    cout << "Pop Operation: " << pop() << endl;
    cout << "Pop Operation after removing \"2\" from the stack: " << top() << endl;

    return 0;
}
